modules/reactor/src/meshgenerators/FlexiblePatternGenerator.C

```diff
@@ -553,9 +556,25 @@ FlexiblePatternGenerator::FlexiblePatternGenerator(const InputParameters & param
 std::unique_ptr<MeshBase>
 FlexiblePatternGenerator::generate()
 {
-  if (isParamValid("external_boundary_id"))
-    MooseMesh::changeBoundaryId(
-        **_build_mesh, OUTER_SIDESET_ID, getParam<boundary_id_type>("external_boundary_id"), false);
+  const auto external_boundary_id =
+      isParamValid("external_boundary_id") ? getParam<boundary_id_type>("external_boundary_id") : 0;
```

Comment:
`0` is a valid boundary id, you can use `OUTER_SIDESET_ID` here so that nothing would go wrong.